### ISSUE TITLE
dat cli freezes when fetching to folder with existing content

### ISSUE THREAD

**Issue by watson** [datproject/dat#565]
When fetching a dat from another person into a folder with a lot files and subfolders, the program seems to stall after printing out `Connecting....`. It's not possible to exit the program with `Ctrl+C` and a normal `kill` doesn't work either. I had to `kill -9` for it to abort.

When fetching the same dat into an empty folder everything worked fine.

There were 187 objects in the root of the folder and 105,628 objects if also counting sub folders. Could be that it was trying to traverse the folders.

this is my dat install:

```
└─┬ dat@11.5.5 
  ├─┬ dat-js@3.8.0 
  │ ├─┬ hyperdrive@7.7.1
  │ │ ├─┬ hypercore@4.14.0
  │ │ │ ├─┬ create-hash@1.1.2
  │ │ │ │ └── sha.js@2.4.8 
  │ │ │ └── hypercore-protocol@4.6.0 
  │ │ └─┬ rabin@1.5.0
  │ │   └─┬ debug@2.3.2 
  │ │     └── ms@0.7.2 
  │ └─┬ hyperdrive-archive-swarm@4.1.8
  │   └─┬ signalhub@4.7.4
  │     └─┬ event-source-stream@1.5.0
  │       └─┬ request@2.78.0
  │         └── form-data@2.1.2 
  ├─┬ discovery-swarm@4.0.2
  │ └─┬ utp-native@1.3.1
  │   ├─┬ prebuild@4.4.0
  │   │ └─┬ node-gyp@3.4.0
  │   │   └── which@1.2.12 
  │   └── readable-stream@2.2.2 
  ├─┬ dns-discovery@5.4.2
  │ └─┬ dns-socket@1.4.2
  │   └── dns-packet@1.1.1 
  └─┬ status-logger@3.0.0 
    └── ansi-diff-stream@1.1.1 
```

**Comment by joehand** [comment]
Do you know if you had previously downloaded a different Dat to that folder? 

I saw similar behavior when running the tests. We had an error that wasn't being handled.

**Comment by joehand** [comment]
This should be fixed in 11.6.0. Let me know if you see this again. Thanks!
